main/chains/blocks/index.ts

```diff
@@ -96,7 +97,7 @@ class BlockMonitor extends EventEmitter {
   }
 
   handleBlock (block: Block) {
-    if (block.number !== this.latestBlock) {
+    if (block && block.number !== this.latestBlock) {
```

Comment:
when can block be undefined? I do think I've seen it a few times but I don't think this should realistically ever happen and I think it would be better to have it cause an error than to fail silently

Reply:
updated to log error in this case